R/class2tree.R

```diff
@@ -152,3 +158,183 @@ taxa2dist <- function(x, varstep = FALSE, check = TRUE, labels) {
     warning("you used 'check=FALSE' and some distances are zero -- was this intended?")
   out
 }
+
+###############################################################################
+#################### GET LIST OF ALL TAXONOMY RANK AND IDs ####################
+###############################################################################
+getRankList <- function(x){
+  rankDf <- x[, 'rank']
+  names(rankDf) <- x[, 'rank']
+
+  idDf <- x[, 'id']
+  joinedDf <- cbind(data.frame(rankDf,stringsAsFactors=FALSE),data.frame(idDf,stringsAsFactors=FALSE))
+  joinedDf <- within(joinedDf, rankDf[rankDf=='no rank'] <- paste0("norank_",idDf[rankDf=='no rank']))
+
+  df <- data.frame(t(data.frame(rev(joinedDf$rankDf))), stringsAsFactors = FALSE)
+  outDf <- data.frame(tip = x[nrow(x), "name"], df, stringsAsFactors = FALSE)
```

Comment:
Is there a reason this fxn and the next below don't explicitly return any output?

Reply:
@sckott  think you don't mean that they should use `return()` statement, but just have implicit return, ie., last line
```r
outDf
```